### ISSUE TITLE
Compiler tests don't report warning for pointer to System.Nullable

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#77725]
Paste in the following test

```cs
[Fact]
public void WRN_ManagedAddr()
{
    var source = """
        class C { }
        struct S()
        {
            public C C = new C();
        }

        unsafe class Program
        {
            static void M(
                S* s, // 1
                S?* ns) // 2
            {
                (*s).C.ToString();
            }
        }
        """;
    var comp = CreateCompilation(source, options: TestOptions.UnsafeDebugDll);
    comp.VerifyEmitDiagnostics(
        // (10,12): warning CS8500: This takes the address of, gets the size of, or declares a pointer to a managed type ('S')
        //         S* s, // 1
        Diagnostic(ErrorCode.WRN_ManagedAddr, "s").WithArguments("S").WithLocation(10, 12),
        // (11,13): warning CS8500: This takes the address of, gets the size of, or declares a pointer to a managed type ('S?')
        //         S?* ns) // 2
        Diagnostic(ErrorCode.WRN_ManagedAddr, "ns").WithArguments("S?").WithLocation(11, 13));
}
```

**Expected**: Test passes under all TFMs
**Actual**: Both diagnostics are reported in net472 while only `// 1` is reported in net9.0

It probably has to do with a difference in the reference assemblies used for these test compilations in each TFM.

The problem doesn't repro in a simple net9 console app.

**Comment by jaredpar** [comment]
>  Both diagnostics are reported in net472 while only // 1 is reported in net9.0

That is because on `net9.0` you are compiling against `netstandard2.0` not `net9.0`. That is how the plumbing works, right or wrong. I've thought about making the _standard_ build jupm between `net472` and .NET Current but haven't gone that far yet. In a simple console app targeting `netstandard2.0` you will see that there is only one warning due to a different ref assembly.
